include/boost/numeric/odeint/stepper/system_categories.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ [auto_generated]
+ boost/numeric/odeint/stepper/system_categories.hpp
+
+ [begin_description]
+ Definition of all system categories.
+ [end_description]
+
+ Copyright 2017 Markus Friedrich
+
+ Distributed under the Boost Software License, Version 1.0.
+ (See accompanying file LICENSE_1_0.txt or
+ copy at http://www.boost.org/LICENSE_1_0.txt)
+ */
+
+
+#ifndef BOOST_NUMERIC_ODEINT_STEPPER_SYSTEM_CATEGORIES_HPP_INCLUDED
+#define BOOST_NUMERIC_ODEINT_STEPPER_SYSTEM_CATEGORIES_HPP_INCLUDED
+
+namespace boost {
+namespace numeric {
+namespace odeint {
+
+
+/*
+ * Tags to specify system types
+ *
+ * These tags can be used to detect which system category is used
+ */
+
+struct explicit_system_tag {};
```

Comment:
It would be great to define the interface expected from the different system functor here!

Reply:
How? Just as (doxygen) comments?

Reply:
Yes, just a short doxy showing how this interface is being called from inside the corresponding stepper.